`gefyra run`: confusing error message when container exits immediately

When the container exits immediately, the error message is not helpful. Additionally the info message still prints just as like it succeeded. For example when running `gefyra run` for the buzzword-counter without specifying a command:
```
~$ gefyra run -i quay.io/blueshoe/buzzword-counter:latest -N buzzword-counter-dev -n default

[ERROR] Gateway patch could not be applied to 'buzzword-counter-dev': b'Cannot open network namespace "0": No such file or directory\nCannot open network namespace "0": No such file or directory\n'
[INFO] Container image 'k3d-buzzword-counter-registry/buzzword-counter:0.1.0, buzzword-counter:dev, k3d-buzzword-counter-registry:43287/buzzword-counter:0.1.0, k3d-buzzword-counter-registry:45319/buzzword-counter:0.1.0' started with name 'buzzword-counter-dev' in Kubernetes namespace 'default'
```

Running the command with the additional `-c "bash -c 'tail -f /dev/null'"` starts the container as expected.

--- a/client/gefyra/configuration.py
+++ b/client/gefyra/configuration.py
@@ -98,6 +98,7 @@ class ClientConfiguration(object):
         self.NETWORK_NAME = network_name or "gefyra"
         self.BRIDGE_TIMEOUT = 60  # in seconds
         self.CARGO_PROBE_TIMEOUT = 10  # in seconds
+        self.CONTAINER_RUN_TIMEOUT = 10  # in seconds
         self.KUBE_CONFIG_FILE = kube_config_file
 
     def _init_docker(self):

--- a/client/gefyra/local/bridge.py
+++ b/client/gefyra/local/bridge.py
@@ -151,6 +151,10 @@ def deploy_app_container(
     container = handle_docker_run_container(config, image, **not_none_kwargs)
 
     cargo = config.DOCKER.containers.get(config.CARGO_CONTAINER_NAME)
+
+    if container.status != "running":
+        raise RuntimeError("Container is not running. Did you miss a valid startup command?")
+
     exit_code, output = cargo.exec_run(
         f"bash patchContainerGateway.sh {container.name} {cargo_ip}"
     )

--- a/client/gefyra/local/bridge.py
+++ b/client/gefyra/local/bridge.py
@@ -153,7 +153,9 @@ def deploy_app_container(
     cargo = config.DOCKER.containers.get(config.CARGO_CONTAINER_NAME)
 
     if container.status != "running":
-        raise RuntimeError("Container is not running. Did you miss a valid startup command?")
+        raise RuntimeError(
+            "Container is not running. Did you miss a valid startup command?"
+        )
 
     exit_code, output = cargo.exec_run(
         f"bash patchContainerGateway.sh {container.name} {cargo_ip}"

--- a/client/gefyra/local/bridge.py
+++ b/client/gefyra/local/bridge.py
@@ -128,6 +128,7 @@ def deploy_app_container(
     auto_remove: bool = None,
     dns_search: str = "default",
 ) -> Container:
+    import docker
 
     gefyra_net = config.DOCKER.networks.get(config.NETWORK_NAME)
 
@@ -151,6 +152,20 @@ def deploy_app_container(
     container = handle_docker_run_container(config, image, **not_none_kwargs)
 
     cargo = config.DOCKER.containers.get(config.CARGO_CONTAINER_NAME)
+
+    # busy wait for the container to start
+    try:
+        _i = 0
+        while container.status == "created" and _i < (
+            config.CONTAINER_RUN_TIMEOUT * 10
+        ):
+            sleep(0.1)
+            container = config.DOCKER.containers.get(container.id)
+            _i = _i + 1
+    except docker.errors.NotFound:
+        raise RuntimeError(
+            "Container is not running. Did you miss a valid startup command?"
+        )
 
     if container.status != "running":
         raise RuntimeError(

